src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/TemplateOptionResult.cs

```diff
@@ -47,7 +47,7 @@ internal class TemplateOptionResult
             return new TemplateOptionResult(
                     option,
                     optionResult.Token?.Value ?? string.Empty,
-                    optionResult.GetValueOrDefault<string>());
+                    option.Option.ValueType == typeof(bool) ? optionResult.GetValueOrDefault<bool>().ToString() : optionResult.GetValueOrDefault<string>());
```

Comment:
does it make sense to make it more generic?
the type may also be `int`, though more rare.

```suggestion
             optionResult.GetValueOrDefault().ToString();
```

not sure if this overload still exists though